Chess-Challenge.Uci/UciEngine.cs

```diff
@@ -0,0 +1,144 @@
+using ChessChallenge.API;
+using ChessChallenge.Chess;
+using Board = ChessChallenge.Chess.Board;
+using Move = ChessChallenge.Chess.Move;
+using Timer = ChessChallenge.API.Timer;
+
+namespace ChessChallenge.Bot;
+
+public class UciEngine
+{
+    private const string UciInit = "uci";
+    private const string UciOkay = "uciok";
+    private const string IsReady = "isready";
+    private const string ReadyOk = "readyok";
+    private const string NewGame = "ucinewgame";
+    private const string Position = "position";
+    private const string BestMove = "bestmove";
+    private const string Go = "go";
+    private const string Stop = "stop";
+    private const string Quit = "quit";
+
+    private const string botMatchStartFens = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";
+
+
+    private MyBot _bot = new MyBot();
+    private Board _currentBoard;
+
+    private string _logFile = "./comm-log.txt";
+    private string _uciLog = "uci-log.txt";
+    private string _errorFile = "error.log";
+    private const string _dateFormat = "yyyyMMddTHHmmss";
+
+
+    private char GetPromotionCharacter(PieceType piece) =>
+        piece switch
+        {
+            PieceType.None => 'q',
```

Comment:
Typo? None and Pawn are both 'q', doesn't look right.

Reply:
ah, that was from testing. We don't need that method anymore, I'll get it pulled out.